scripts/build-plugins/rollup-plugin-build-themes.js

```diff
@@ -254,13 +254,50 @@ module.exports = function buildThemes(options) {
                 const derivedVariables = compiledVariables["derived-variables"];
                 const icon = compiledVariables["icon"];
                 const builtAssets = {};
-                /**
-                 * Generate a mapping from theme name to asset hashed location of said theme in build output.
-                 * This can be used to enumerate themes during runtime.
-                 */
+                let defaultDarkVariant = {}, defaultLightVariant = {};
+                const themeName = manifest.name;
                 for (const chunk of chunkArray) {
                     const [, name, variant] = chunk.fileName.match(/theme-(.+)-(.+)\.css/);
-                    builtAssets[`${name}-${variant}`] = assetMap.get(chunk.fileName).fileName;
+                    const {name: variantName, default: isDefault, dark} = manifest.values.variants[variant];
+                    const themeId = `${name}-${variant}`;
+                    const themeDisplayName = `${themeName} ${variantName}`;
+                    if (isDefault) {
+                        /**
+                         * This is a default variant!
+                         * We'll add these to the builtAssets keyed with just the
+                         * theme-name (i.e "Element" instead of "Element Dark")
+                         * so that we can distinguish them from other variants!
+                         * 
+                         * This allows us to render a radio-button with "dark" and
+                         * "light" options.
+                        */ 
+                        const defaultVariant = dark ? defaultDarkVariant : defaultLightVariant;
+                        defaultVariant.themeDisplayName = variantName;
+                        defaultVariant.id = themeId
+                        defaultVariant.cssLocation = assetMap.get(chunk.fileName).fileName;
+                        continue;
+                    }
+                    // Non-default variants are keyed in builtAssets with "theme_name variant_name"
+                    // eg: "Element Dark"
+                    builtAssets[themeDisplayName] = {
```

Comment:
Why not just map the variant id (which we know is unique, as it's used as a key) to the css file here? Are you perhaps trying to have a structure in the manifest that maps 1:1 to how we want to present themes in the settings UI? I would go for a format without any duplication (and mangle the data to present in the settings UI at runtime), so for `built-assets` instead of this: 
```json
{
            "Element Violet":
            {
                "cssLocation": "assets/theme-element-violet.d45003f1.css",
                "id": "element-violet"
            },
            "Element":
            {
                "dark":
                {
                    "variantName": "Dark",
                    "id": "element-dark",
                    "cssLocation": "assets/theme-element-dark.2f1cd6cb.css"
                },
                "light":
                {
                    "variantName": "Light",
                    "id": "element-light",
                    "cssLocation": "assets/theme-element-light.a49cf48a.css"
                }
            }
        }
```
..., just this?
```json
{
    "violet": "assets/theme-element-violet.d45003f1.css",
    "dark": "assets/theme-element-dark.2f1cd6cb.css",
    "light": "assets/theme-element-light.a49cf48a.css"
}